qunicorn_core/api/api_models/result_dtos.py

```diff
@@ -33,14 +33,14 @@
 class ResultDto:
     id: int
     circuit: Optional[str]
-    result_dict: dict
-    meta_data: dict
+    data: dict
```

Comment:
```suggestion
    data: dict | list | tuple | str | float | int | bool
```
We might have data that is not a dictionary in the future (e.g. a single expectation value).
(alternatively use `Any` to express that the exact type is unknown for now)